File: XmlLogger/XmlLogger/BuildResult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Text;
using System.Threading.Tasks;

namespace XmlLogger
{
    [DataContract(Name = "buildresult")]
    internal class BuildResult
    {
        [DataMember(Name = "successful", Order = 0)]
        public bool Successful;

        [DataMember(Name = "errorlist", Order = 1)]
        public List<Error> ErrorList;

        public BuildResult()
        {
            ErrorList = new List<Error>();
        }
    }
}

File: src/server/buildresult.go
package main

type Error struct {
	Type string `xml:"type"`

	FileName     string `xml:"filename"`
	LineNumber   int    `xml:"linenumber"`
	ColumnNumber int    `xml:"columnnumber"`

	Code    string `xml:"code"`
	Message string `xml:"message"`
}

type BuildResult struct {
	Successful bool `xml:"successful"`

	ErrorList []Error `xml:"errorlist>error"`
}
